File: admin-panel/src/config/env.ts
/**
 * Environment configuration loaded from Vite's import.meta.env
 *
 * Values are injected from .env.local during build.
 * See: scripts/deploy/generate-env.ps1
 *
 * Usage:
 * ```typescript
 * import { env } from '@/config/env';
 *
 * const url = env.supabaseUrl;
 * const version = env.appVersion;
 * ```
 */

interface EnvConfig {
  /** Application version from master-config.json */
  appVersion: string;
  /** Application display name */
  appName: string;
  /** Supabase project URL */
  supabaseUrl: string;
  /** Supabase anonymous key (safe for client-side) */
  supabaseAnonKey: string;
  /** Enable offline mode features */
  enableOfflineMode: boolean;
  /** Analytics tracking ID (optional) */
  analyticsId: string | null;
  /** Gemini API key for AI features (optional) */
  geminiApiKey: string | null;
  /** Current environment mode */
  mode: 'development' | 'production';
  /** Whether running in production */
  isProduction: boolean;
  /** Whether running in development */
  isDevelopment: boolean;
}

/**
 * Get an environment variable with optional required check.
 *
 * @param key - The VITE_ prefixed environment variable key
 * @param required - Whether to throw if the variable is missing
 * @returns The environment variable value or empty string
 */
function getEnvVar(key: string, required = true): string {
  const value = import.meta.env[key];
  if (required && !value) {
    console.error(`Missing required environment variable: ${key}`);
    // In development, show a warning but don't crash
    if (import.meta.env.DEV) {
      console.warn(`Continuing with empty value for ${key} in development mode`);
      return '';
    }
    throw new Error(`Missing required environment variable: ${key}`);
  }
  // Convert to string since index signature can return boolean
  return typeof value === 'string' ? value : '';
}

/**
 * Environment configuration object.
 * All values are loaded at module initialization time.
 */
export const env: EnvConfig = {
  appVersion: getEnvVar('VITE_APP_VERSION', false) || '0.0.0',
  appName: getEnvVar('VITE_APP_NAME', false) || 'Questerix Admin',
  supabaseUrl: getEnvVar('VITE_SUPABASE_URL'),
  supabaseAnonKey: getEnvVar('VITE_SUPABASE_ANON_KEY'),
  enableOfflineMode: getEnvVar('VITE_ENABLE_OFFLINE_MODE', false) === 'true',
  analyticsId: getEnvVar('VITE_ANALYTICS_ID', false) || null,
  geminiApiKey: getEnvVar('VITE_GEMINI_API_KEY', false) || null,
  mode: import.meta.env.MODE as 'development' | 'production',
  isProduction: import.meta.env.PROD,
  isDevelopment: import.meta.env.DEV,
};

/**
 * Validate that all required environment variables are set.
 * Call this at app startup to fail fast.
 *
 * @throws Error if required variables are missing
 */
export function validateEnv(): void {
  const missing: string[] = [];

  if (!env.supabaseUrl) missing.push('VITE_SUPABASE_URL');
  if (!env.supabaseAnonKey) missing.push('VITE_SUPABASE_ANON_KEY');

  if (missing.length > 0) {
    throw new Error(
      `Missing required environment variables: ${missing.join(', ')}\n` +
        'Ensure .env.local is properly configured.\n' +
        'Run: ./scripts/deploy/generate-env.ps1 to generate from master-config.json'
    );
  }
}

// Log configuration in development (with sensitive values redacted)
if (import.meta.env.DEV) {
  console.log('[ENV] Loaded configuration:', {
    appVersion: env.appVersion,
    appName: env.appName,
    mode: env.mode,
    supabaseUrl: env.supabaseUrl ? '***configured***' : 'NOT SET',
    supabaseAnonKey: env.supabaseAnonKey ? '***configured***' : 'NOT SET',
    enableOfflineMode: env.enableOfflineMode,
    analyticsId: env.analyticsId ? '***configured***' : null,
    geminiApiKey: env.geminiApiKey ? '***configured***' : null,
  });
}

File: student-app/lib/src/core/config/env.dart
/// Environment configuration loaded from compile-time dart-define values.
///
/// These values are injected during `flutter build` via --dart-define flags.
/// See: scripts/deploy/build-student.ps1
///
/// Usage:
/// ```dart
/// import 'package:student_app/src/core/config/env.dart';
///
/// final url = Env.supabaseUrl;
/// final version = Env.appVersion;
/// ```
///
/// To add new environment variables:
/// 1. Add the variable to master-config.json under "student" section
/// 2. Add the --dart-define flag in generate-env.ps1
/// 3. Add the accessor in this file
class Env {
  Env._(); // Prevent instantiation

  /// Application version (synced with master-config.json)
  static const String appVersion = String.fromEnvironment(
    'APP_VERSION',
    defaultValue: 'dev',
  );

  /// Application display name
  static const String appName = String.fromEnvironment(
    'APP_NAME',
    defaultValue: 'Questerix',
  );

  /// Supabase project URL
  static const String supabaseUrl = String.fromEnvironment(
    'SUPABASE_URL',
    defaultValue: '',
  );

  /// Supabase anonymous key (safe for client-side use)
  static const String supabaseAnonKey = String.fromEnvironment(
    'SUPABASE_ANON_KEY',
    defaultValue: '',
  );

  /// Primary theme color as hex string (e.g., "0xFF319795")
  static const String _themePrimaryColorRaw = String.fromEnvironment(
    'THEME_PRIMARY_COLOR',
    defaultValue: '0xFF319795',
  );

  /// Parsed theme color as integer
  static int get themePrimaryColor => int.parse(_themePrimaryColorRaw);

  /// Enable offline-first mode with Drift
  static const String _offlineFirstRaw = String.fromEnvironment(
    'ENABLE_OFFLINE_FIRST',
    defaultValue: 'true',
  );

  /// Whether offline-first mode is enabled
  static bool get enableOfflineFirst =>
      _offlineFirstRaw.toLowerCase() == 'true';

  /// Drift database schema version
  static const String _driftDbVersionRaw = String.fromEnvironment(
    'DRIFT_DB_VERSION',
    defaultValue: '1',
  );

  /// Parsed Drift database version
  static int get driftDbVersion => int.parse(_driftDbVersionRaw);

  /// Current environment (development, staging, production)
  static const String environment = String.fromEnvironment(
    'ENV',
    defaultValue: 'development',
  );

  /// Check if running in production
  static bool get isProduction => environment == 'production';

  /// Check if running in development
  static bool get isDevelopment => environment == 'development';

  /// Validate that required environment variables are set.
  ///
  /// Call this at app startup to fail fast if configuration is missing.
  /// Throws [StateError] if required variables are not set.
  ///
  /// Example:
  /// ```dart
  /// void main() {
  ///   Env.validate(); // Throws if SUPABASE_URL or SUPABASE_ANON_KEY missing
  ///   runApp(MyApp());
  /// }
  /// ```
  static void validate() {
    final missing = <String>[];

    if (supabaseUrl.isEmpty) missing.add('SUPABASE_URL');
    if (supabaseAnonKey.isEmpty) missing.add('SUPABASE_ANON_KEY');

    if (missing.isNotEmpty) {
      throw StateError(
        'Missing required environment variables: ${missing.join(', ')}\n'
        'Ensure flutter build includes --dart-define flags.\n'
        'Run: ./scripts/deploy/generate-env.ps1 before building.',
      );
    }
  }

  /// Get a summary of the current environment configuration.
  /// Useful for debugging (masks sensitive values).
  static Map<String, String> get summary => {
        'appVersion': appVersion,
        'appName': appName,
        'environment': environment,
        'supabaseUrl': supabaseUrl.isNotEmpty ? '***configured***' : 'NOT SET',
        'supabaseAnonKey':
            supabaseAnonKey.isNotEmpty ? '***configured***' : 'NOT SET',
        'themePrimaryColor': _themePrimaryColorRaw,
        'enableOfflineFirst': enableOfflineFirst.toString(),
        'driftDbVersion': driftDbVersion.toString(),
      };
}
